pkg/evetestkit/utils.go

```diff
@@ -0,0 +1,487 @@
+		Timeout:         30 * time.Second,
+	}
+	client, err := ssh.Dial("tcp", host, config)
+	if err != nil {
+		return fmt.Errorf("failed to dial: %s", err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	err = scp.CopyPath(localFile, remoteFile, session)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// WithSSH is an option that sets the SSH configuration for the app running on
+// the EVE node, this should be use with DeployVM function.
+func WithSSH(user, pass, port string) VMOption {
+	return func(n *EveNode) {
+		// slices are ordered, so the last element is the one
+		// we're going to set things for.
+		n.apps[len(n.apps)-1].sshUser = user
+		n.apps[len(n.apps)-1].sshPass = pass
+		n.apps[len(n.apps)-1].sshPort = port
+	}
+}
+
+// DeployVm deploys a VM on the EVE node
+func (node *EveNode) DeployVM(imageURL string, pc openevec.PodConfig, options ...VMOption) error {
+	node.apps = append(node.apps, appInstanceConfig{name: pc.Name})
+	for _, option := range options {
+		option(node)
+	}
+
+	return node.controller.PodDeploy(imageURL, pc, node.cfg)
+}
+
+// IsTpmEnabled checks if EVE node is running with (SW)TPM enabled
+func (node *EveNode) IsTpmEnabled() bool {
+	return node.cfg.Eve.TPM
+}
+
+func (node *EveNode) setEveIP() error {
+	if node.edgenode.GetRemoteAddr() == "" {
+		eveIPCIDR, err := node.tc.GetState(node.edgenode).LookUp("Dinfo.Network[0].IPAddrs[0]")
+		if err != nil {
+			return err
+		}
+
+		ip := net.ParseIP(eveIPCIDR.String())
+		if ip == nil || ip.To4() == nil {
+			return fmt.Errorf("failed to parse IP address: %s", eveIPCIDR.String())
+		}
+
+		node.ip = ip.To4().String()
+		return nil
+	}
+
+	node.ip = node.edgenode.GetRemoteAddr()
+	return nil
+}
+
+// GetDefaultVMConfig returns the default configuration for a VM
+func GetDefaultVMConfig(appName, cloudConfig string, portPub []string) openevec.PodConfig {
+	var pc openevec.PodConfig
+
+	pc.Name = appName
+	pc.AppMemory = humanize.Bytes(defaults.DefaultAppMem * 1024)
+	pc.DiskSize = "4GB"
+	pc.VolumeType = "QCOW2"
+	pc.Metadata = cloudConfig
+	pc.VncPassword = ""
+	pc.ImageFormat = "QCOW2"
+	pc.Registry = "remote"
+	pc.VolumeSize = humanize.IBytes(defaults.DefaultVolumeSize)
+	pc.PortPublish = portPub
+	pc.VncDisplay = 0
+	pc.AppCpus = defaults.DefaultAppCPU
+	pc.AppAdapters = nil
+	pc.Networks = nil
+	pc.ACLOnlyHost = false
+	pc.NoHyper = false
+	pc.DirectLoad = true
+	pc.SftpLoad = false
+	pc.Disks = nil
+	pc.Mount = nil
+	pc.Profiles = nil
+	pc.ACL = nil
+	pc.Vlans = nil
+	pc.OpenStackMetadata = false
+	pc.DatastoreOverride = ""
+	pc.StartDelay = 0
+	pc.PinCpus = false
+
+	return pc
+}
+
+// SetControllerVerbosity sets the verbosity level of the controller,
+// possible values are: panic, fatal, error, debug, info, trace, warn
+// this should be set before calling InitilizeTest.
+func SetControllerVerbosity(verbosity string) {
```

Comment:
`SetControllerVerbosity` and `SetEdenConfigEnv` could be options for `InitilizeTest`